Subject: Linux-RT 2.6.25.4-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From sshtylyov@ru.mvista.com Thu May 17 15:18:39 2007
Return-Path: <sshtylyov@ru.mvista.com>
X-Spam-Checker-Version: SpamAssassin 3.1.7-deb (2006-10-05) on debian
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=AWL autolearn=unavailable 
	version=3.1.7-deb
Received: from imap.sh.mvista.com (unknown [63.81.120.155]) by mail.tglx.de
	(Postfix) with ESMTP id BFD3A65C065 for <tglx@linutronix.de>; Thu, 17 May
	2007 15:18:39 +0200 (CEST)
Received: from wasted.dev.rtsoft.ru (unknown [10.150.0.9]) by
	imap.sh.mvista.com (Postfix) with ESMTP id 8E3CB3EC9; Thu, 17 May 2007
	06:18:35 -0700 (PDT)
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Organization: MontaVista Software Inc.
To: mingo@elte.hu, tglx@linutronix.de
Subject: [PATCH 2.6.21-rt2] PowerPC: revert fix for threaded fasteoi IRQ
	handlers
Date: Thu, 17 May 2007 17:20:08 +0400
User-Agent: KMail/1.5
Cc: linux-kernel@vger.kernel.org, linuxppc-dev@ozlabs.org, dwalker@mvista.com
References: <200611192243.34850.sshtylyov@ru.mvista.com>
In-Reply-To: <200611192243.34850.sshtylyov@ru.mvista.com>
MIME-Version: 1.0
Content-Disposition: inline
Message-Id: <200705171719.34968.sshtylyov@ru.mvista.com>
Content-Type: text/plain; charset="us-ascii"
X-Evolution-Source: imap://tglx%40linutronix.de@localhost:8993/
Content-Transfer-Encoding: 8bit

Revert the change to the "fasteoi" type chips as after handle_fasteoi_irq() had
been fixed, they've become meaningless (and even dangerous -- as was the case
with Celleb that has been fixed earlier)...

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>

---
The patch in question wasn't even initially accepted but then was erroneously
restored along with the TOD patch.  I've asked to revert it but to no avail,
so here's the formal patch to revert it at last...

 arch/powerpc/platforms/iseries/irq.c  |    1 -
 arch/powerpc/platforms/pseries/xics.c |    2 --
 arch/powerpc/sysdev/mpic.c            |    1 -
 3 files changed, 4 deletions(-)

Index: linux-2.6.25.4-rt2/arch/powerpc/platforms/iseries/irq.c
===================================================================
--- linux-2.6.25.4-rt2.orig/arch/powerpc/platforms/iseries/irq.c	2008-05-19 16:55:35.000000000 -0400
+++ linux-2.6.25.4-rt2/arch/powerpc/platforms/iseries/irq.c	2008-05-19 16:55:36.000000000 -0400
@@ -278,7 +278,6 @@ static struct irq_chip iseries_pic = {
 	.shutdown	= iseries_shutdown_IRQ,
 	.unmask		= iseries_enable_IRQ,
 	.mask		= iseries_disable_IRQ,
-	.ack		= iseries_end_IRQ,
 	.eoi		= iseries_end_IRQ
 };
 
Index: linux-2.6.25.4-rt2/arch/powerpc/platforms/pseries/xics.c
===================================================================
--- linux-2.6.25.4-rt2.orig/arch/powerpc/platforms/pseries/xics.c	2008-05-19 16:55:35.000000000 -0400
+++ linux-2.6.25.4-rt2/arch/powerpc/platforms/pseries/xics.c	2008-05-19 16:55:36.000000000 -0400
@@ -502,7 +502,6 @@ static struct irq_chip xics_pic_direct =
 	.startup = xics_startup,
 	.mask = xics_mask_irq,
 	.unmask = xics_unmask_irq,
-	.ack = xics_eoi_direct,
 	.eoi = xics_eoi_direct,
 	.set_affinity = xics_set_affinity
 };
@@ -513,7 +512,6 @@ static struct irq_chip xics_pic_lpar = {
 	.startup = xics_startup,
 	.mask = xics_mask_irq,
 	.unmask = xics_unmask_irq,
-	.ack = xics_eoi_lpar,
 	.eoi = xics_eoi_lpar,
 	.set_affinity = xics_set_affinity
 };
Index: linux-2.6.25.4-rt2/arch/powerpc/sysdev/mpic.c
===================================================================
--- linux-2.6.25.4-rt2.orig/arch/powerpc/sysdev/mpic.c	2008-05-19 16:55:35.000000000 -0400
+++ linux-2.6.25.4-rt2/arch/powerpc/sysdev/mpic.c	2008-05-19 16:55:36.000000000 -0400
@@ -864,7 +864,6 @@ void mpic_set_vector(unsigned int virq, 
 static struct irq_chip mpic_irq_chip = {
 	.mask		= mpic_mask_irq,
 	.unmask		= mpic_unmask_irq,
-	.ack		= mpic_end_irq,
 	.eoi		= mpic_end_irq,
 	.set_type	= mpic_set_irq_type,
 };
